test/e2e/testsuites/scale_tester.go

```diff
@@ -0,0 +1,342 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package testsuites
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	scale "k8s.io/api/autoscaling/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"sigs.k8s.io/azuredisk-csi-driver/test/e2e/driver"
+)
+
+type PodSchedulingWithPVScaleTest struct {
+	CSIDriver              driver.DynamicPVTestDriver
+	Pods                   []PodDetails
+	PodCount               int
+	StorageClassParameters map[string]string
+}
+
+type PodSchedulingOnFailoverScaleTest struct {
+	CSIDriver              driver.DynamicPVTestDriver
+	Pods                   []PodDetails
+	PodCount               int
+	StorageClassParameters map[string]string
+}
+
+type PodSchedulingWithMultiplePVScaleTest struct {
+	CSIDriver driver.DynamicPVTestDriver
+	Pods      []PodDetails
+}
+
+type PodSchedulingOnFailoverMultiplePVScaleTest struct {
+	CSIDriver              driver.DynamicPVTestDriver
+	Pods                   []PodDetails
+	Replicas               int
+	StorageClassParameters map[string]string
+}
+
+func (t *PodSchedulingWithPVScaleTest) Run(client clientset.Interface, namespace *v1.Namespace, schedulerName string) {
+	nodeNames := ListNodeNames(client)
+	if len(nodeNames) < 1 {
```

Comment:
1 is a really small number for a scale test. In order for this check to be effective, this number has to reflect the min number of pods necessary to accommodate the test.

Reply:
Resolved for all the Run functions